### DISCUSSION TITLE
Traefik Pilot intergration??

### DISCUSSION THREAD

**Discussion by johnkennyuk** [demyxco/demyx#16] [Ideas]
I tried to get Traefik Pilot to work with Demyx/Traefik but i assume docker socket app was getting in the way of the cloud app to the local demyx/traefik instance??  I used the TRAEFIK_PILOT_TOKEN=Value Key Value in the ENV file as well as in the Traefik Docker-compose.yml file as both ENV variable and Command attribute??  I will continue to troubleshoot the token connection issue, but it would be ideal to have it added as an option via the cli cmds???  It would certainly allow quick injection of different MiddleWares such as Cloud Authentication type MiddleWares i.e. google oauth or Microsoft AAD Oauth??

Would be a great addition which would also go great with Traefik V2.4??

Great app Demyx one the only apps that can deploy Traffik Infront of WordPress instances in the case of Demyx and hopefully other apps to come?

**Comment by demyxco** [comment]
Hi @johnkennyuk the demyx stack is pretty much fixed at the moment. A few ways modify them would be forking demyx or building your own Dockerfile and base it off of any of the demyx images.

I've never used Trust Pilot but the `demyx/traefik` container is configured to use the `demyx_socket` network. So I'm assuming Trust Pilot needs access to the docker.sock right? If so, then you'll need to define the `demyx_socket` network in there and set the environment variable to `DOCKER_HOST=tcp://demyx_socket:2375`.

**Reply by johnkennyuk** [reply]
unfortunately there isn't much documentation for Traefik Pilot, I'll need to do some digging but because it only requires a Token to connect I'm assuming Contanious built in something for Traffic Pilot in Traffic compatible versions.  But I'll keep digging and if I find anything ill reply here.

As you can see little content, but it seems Traefik connects out not in??
https://doc.traefik.io/traefik-pilot/connecting/

All were told is if your local Traefik instance has the generated Pilot Token and its added via CLI, ENV or Static configs it should work?? Ive had it working standalone but not managed to do so using the Demyx Traefik app again I'm assuming it's because Docker Socket is inbetween Traefik and Traefik Pilot?? Would that be the case as i don't know much about Docker Socket??

**Reply by demyxco** [reply]
I've read it up a little bit and it looks like it manages multiple Traefik instances, assuming this would require you to open up the docker socket port. It might be out of scope for demyx at the moment.
